persistence/persistence-mybatis/src/main/java/org/cbioportal/persistence/mybatis/AlterationMyBatisRepository.java

```diff
@@ -56,7 +59,15 @@
         Map<MolecularAlterationType, List<MolecularProfileCaseIdentifier>> groupedIdentifiersByProfileType =
             alterationCountsMapper.getMolecularProfileCaseInternalIdentifier(new ArrayList<>(molecularProfileCaseIdentifiers), "SAMPLE_ID")
             .stream()
-            .collect(Collectors.groupingBy(e -> profileTypeByProfileId.getOrDefault(e.getMolecularProfileId(), null)));
+            .peek(e -> {
+                // Collectors.groupingBy does not allow null keys (it NPEs). The keys for this groupBy should never be null,
+                // but if they are, we filter them out and log a warning about them, rather than letting this whole function break.
+                if (!profileTypeByProfileId.containsKey(e.getMolecularProfileId())) {
+                    LOG.warn("Could not find internal molecular profile id {} - you probably need to clear your cache or restart your server.", e.getMolecularProfileId());
+                }
+            })
+            .filter(e -> profileTypeByProfileId.containsKey(e.getMolecularProfileId()))
+            .collect(Collectors.groupingBy(e -> profileTypeByProfileId.get(e.getMolecularProfileId())));
```

Comment:
```suggestion
            .collect(Collectors.groupingBy(e -> profileTypeByProfileId.get(e.getMolecularProfileId())));
```

@inodb  we could just do this instead. This will still NPE, but at least we aren't explicitly trying to group by null

Reply:
I like the explicit message that indicates what to do, but maybe exception is better? Just because I guess the numbers would be wrong and people might draw wrong conclusions